src/Lean/Parser/Types.lean

```diff
@@ -365,7 +365,11 @@ def adaptUncacheableContextFn (f : ParserContextCore → ParserContextCore) (p :
 
 /--
 Run `p` and record result in parser cache for any further invocation with this `parserName`, parser context, and parser state.
-`p` cannot access syntax stack elements pushed before the invocation in order to make caching independent of parser history. -/
+`p` cannot access syntax stack elements pushed before the invocation in order to make caching independent of parser history.
+As this excludes trailing parsers from being cached, we also reset `lhsPrec`, which is not read but set by leading parsers, to 0
+in order to increase cache hits. Finally, `errorMsg` is also reset to `none` as a leading parser should not be called in the first
+place if there was an error.
```

Comment:
Sounds reasonable? Test suite says "yes". All `ParserState` fields are accounted for now.

Reply:
I think so. It is great you are documenting it.